### ISSUE TITLE
JSDoc param for function overload parameter is incorrectly marked as unused

### ISSUE THREAD

**Issue by mjbvz** [microsoft/TypeScript#49469]
# Bug Report

### 🔎 Search Terms
- jsdoc
- overload
- function signature 

### 🕗 Version & Regression Information

4.8.0-dev.20220606, not a regression 

### 💻 Code

```ts
/**
 * @param bla A bla
 */
function foo(): void;
function foo(bla?: number): void { }
```

### 🙁 Actual behavior

`bla` has a hint saying: `JSDoc '@param' tag has name 'bla', but there is no parameter with that name.`

### 🙂 Expected behavior

`bla` should apply to the `bla` in the overload signature

**Comment by RyanCavanaugh** [comment]
Two reasons this is wrong - doc comments are supposed to be above the signature they apply to, and the implementation signature isn't externally visible, so there's nothing to document.
